src/Integration/DeprecationManager.cs

```diff
@@ -0,0 +1,88 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration.InfoBar;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    public sealed class DeprecationManager : IDisposable
+    {
+        private const string VisualStudio2015Update3VersionString = "14.0.25420.00";
+        private static readonly Version VisualStudio2015Update3Version = Version.Parse(VisualStudio2015Update3VersionString);
+        internal /* for testing purpose */ static readonly Guid DeprecationBarGuid = new Guid(ToolWindowGuids80.ErrorList);
+
+        private readonly IInfoBarManager infoBarManager;
+        private readonly ISonarLintOutput sonarLintOutput;
+        private IInfoBar deprecationBar;
+
+        public DeprecationManager(IInfoBarManager infoBarManager, ISonarLintOutput sonarLintOutput)
+        {
+            if (infoBarManager == null)
+            {
+                throw new ArgumentNullException(nameof(infoBarManager));
+            }
+            if (sonarLintOutput == null)
+            {
+                throw new ArgumentNullException(nameof(sonarLintOutput));
+            }
+
+            this.infoBarManager = infoBarManager;
+            this.sonarLintOutput = sonarLintOutput;
+        }
+
+        public void Initialize(string visualStudioVersion)
+        {
+            Version vsVersion;
+            if (Version.TryParse(visualStudioVersion, out vsVersion) &&
+                vsVersion < VisualStudio2015Update3Version)
+            {
+                WriteMessageToOutput();
+                ShowDeprecationBar();
+            }
+        }
+
+        private void WriteMessageToOutput()
+        {
+            const string message =
+                "*****************************************************************************************\r\n" +
+                "***   Newer versions of SonarLint will not work with this version of Visual Studio.   ***\r\n" +
+                "***   Please update to Visual Studio 2015 Update 3 or Visual Studio 2017 to benefit   ***\r\n" +
+                "***   from new features.                                                              ***\r\n" +
+                "*****************************************************************************************";
+
+            sonarLintOutput.Write(message);
+        }
+
+        private void ShowDeprecationBar()
+        {
+            const string message = "Newer versions of SonarLint will not work with this version of Visual Studio.";
```

Comment:
Please also feel free to suggest changes to this message.

Reply:
The message does not say how to fix that. As a user, I would have thought this means I need to move to VS 2017.

Perhaps use the same message as in output (minus asterisks)?